inc/class-rest-controller.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace ModularPageBuilder;
+
+use WP_REST_Server;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Error;
+use WP_Http;
+use WP_REST_Posts_Controller;
+
+/**
+ * REST API Controller for the page builder routes.
+ */
+class REST_Controller extends \WP_REST_Controller {
+
+	public function __construct( Builder $builder ) {
+		$this->builder = $builder;
+	}
+
+	public function register_routes() {
+		register_rest_route( 'modular-page-builder/v1', $this->builder->args['api_prop'] . '/(?P<id>[\d]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'callback'            => array( $this, 'get_items' ),
+			),
+		) );
+
+		register_rest_route( 'modular-page-builder/v1', $this->builder->args['api_prop'] . '/(?P<id>[\d]+)/(?P<number>[\d]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'callback'            => array( $this, 'get_item' ),
+				'args'                => array(
+					'number'          => array(
+						'sanitize_callback' => 'absint',
+					),
+				),
+			),
+		) );
+	}
+
+	public function get_item_permissions_check( $request ) {
+		$rest_controller = new WP_REST_Posts_Controller( get_post_type( $request['id'] ) );
+
+		return $rest_controller->get_item_permissions_check( $request );
+	}
+
+	public function get_items( $request ) {
+		$modules = $this->builder->get_raw_data( $request['id'] );
+
+		$num = 0;
+
+		$data = array_map( function ( $module ) use ( $request, &$num ) {
+			$module = Plugin::get_instance()->init_module( $module['name'], $module );
+			$response = $this->prepare_item_for_response( $module, $request );
+			$data = $response->get_data();
+			$data['id'] = $num;
+			$num++;
```

Comment:
Could do the `++` on the line above as it'll happen after the value is read though I guess the intent is a bit less obvious then.